transmart-batch/src/main/groovy/org/transmartproject/batch/concept/ConceptTree.groovy

```diff
@@ -115,6 +115,12 @@ class ConceptTree {
         if (node) {
             return node
         }
+        if(ClinicalVariable.conceptTypeFor(variable) != ConceptType.CATEGORICAL) {
```

Comment:
I think that generating missing parent nodes is also okay for categorical variables, even though they would have been added later by `getOrGenerateConceptNode` in `ClinicalFactsRowSetFactory`, especially because that only happens for 'non-strict' categorical nodes, meaning categorical nodes with value nodes below them.
So, the `if` is not needed here.